chain/network/src/test_loop.rs

```diff
@@ -1,4 +1,301 @@
+            let _ = sender.send_async(BlockApproval(approval_message.approval, PeerId::random()));
+            None
+        }
+        NetworkRequests::ForwardTx(account, transaction) => {
+            assert!(account != my_account_id, "Sending message to self not supported.");
+            let sender = client_senders.get(&account).unwrap();
+            let _ = sender.send_async(ProcessTxRequest {
+                transaction,
+                is_forwarded: true,
+                check_only: false,
+            });
+            None
+        }
+        NetworkRequests::ChunkEndorsement(target, endorsement) => {
+            assert!(target != my_account_id, "Sending message to self not supported.");
+            let sender = client_senders.get(&target).unwrap();
+            let _ = sender.send_async(ChunkEndorsementMessage(endorsement));
+            None
+        }
+        _ => Some(request),
+    })
+}
+
+fn network_message_to_partial_witness_handler(
+    my_account_id: &AccountId,
+    partial_witness_senders: HashMap<AccountId, PartialWitnessSenderForNetwork>,
+) -> NetworkRequestHandler {
+    let my_account_id = my_account_id.clone();
+    Arc::new(move |request| match request {
+        NetworkRequests::ChunkStateWitnessAck(target, witness_ack) => {
+            // TODO(testloop): Convert account check to a panic.
+            if target != my_account_id {
+                let sender = partial_witness_senders.get(&target).unwrap();
+                sender.send(ChunkStateWitnessAckMessage(witness_ack));
+            }
+            None
+        }
+
+        NetworkRequests::PartialEncodedStateWitness(validator_witness_tuple) => {
+            for (target, partial_witness) in validator_witness_tuple.into_iter() {
+                // TODO(testloop): Convert account check to a panic.
+                if target != my_account_id {
+                    let sender = partial_witness_senders.get(&target).unwrap();
+                    sender.send(PartialEncodedStateWitnessMessage(partial_witness));
+                }
+            }
+            None
+        }
+        NetworkRequests::PartialEncodedStateWitnessForward(chunk_validators, partial_witness) => {
+            for target in chunk_validators {
+                // TODO(testloop): Convert account check to a panic.
+                if target != my_account_id {
+                    let sender = partial_witness_senders.get(&target).unwrap();
+                    sender.send(PartialEncodedStateWitnessForwardMessage(partial_witness.clone()));
+                }
+            }
+            None
+        }
+        _ => Some(request),
+    })
+}
+
+fn network_message_to_state_snapshot_handler() -> NetworkRequestHandler {
+    Arc::new(move |request| match request {
+        NetworkRequests::SnapshotHostInfo { .. } => None,
+        _ => Some(request),
+    })
+}
+
+#[derive(Default, Clone)]
+struct RouteLookup(Arc<Mutex<HashMap<CryptoHash, AccountId>>>);
```

Comment:
Could you add comment what it stores and does? From context of the PR I have no idea what is `route_id` and why it is needed.